internal/factsengine/gatherers/saplocalhostresolver.go

```diff
@@ -0,0 +1,154 @@
+)
+
+// nolint:gochecknoglobals
+var (
+	SapLocalhostResolverHostnameResolutionError = entities.FactGatheringError{
+		Type:    "saplocalhost_resolver-resolution-error",
+		Message: "error resolving hostname",
+	}
+	SapLocalhostResolverGathererDecodingError = entities.FactGatheringError{
+		Type:    "saplocalhost_resolver-decoding-error",
+		Message: "error decoding output to FactValue",
+	}
+	SapLocalhostResolverFileSystemError = entities.FactGatheringError{
+		Type:    "saplocalhost_resolver-file-system-error",
+		Message: "error reading the sap profiles file system",
+	}
+)
+
+type SapLocalhostResolverGatherer struct {
+	fs afero.Fs
+	hr utils.HostnameResolver
+}
+
+type ResolvabilityDetails struct {
+	Hostname     string   `json:"hostname"`
+	Addresses    []string `json:"addresses"`
+	InstanceName string   `json:"instance_name"`
+}
+
+func NewDefaultSapLocalhostResolverGatherer() *SapLocalhostResolverGatherer {
+	return NewSapLocalhostResolver(afero.NewOsFs(), utils.Resolver{})
+}
+
+func NewSapLocalhostResolver(fs afero.Fs, hr utils.HostnameResolver) *SapLocalhostResolverGatherer {
+	return &SapLocalhostResolverGatherer{fs: fs, hr: hr}
+}
+
+func (r *SapLocalhostResolverGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := make([]entities.Fact, 0, len(factsRequests))
+
+	details, err := r.getInstanceHostnameDetails()
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	for _, factReq := range factsRequests {
+		var fact entities.Fact
+		factValue, err := mapReachabilityDetailsToFactValue(details)
+		if err != nil {
+			log.Error(err)
+			fact = entities.NewFactGatheredWithError(factReq, SapLocalhostResolverGathererDecodingError.Wrap(err.Error()))
+		} else {
+			fact = entities.NewFactGatheredWithRequest(factReq, factValue)
+		}
+		facts = append(facts, fact)
+	}
+
+	return facts, nil
+}
+
+func (r *SapLocalhostResolverGatherer) getInstanceHostnameDetails() (map[string][]ResolvabilityDetails, error) {
+	systems, err := sapsystem.FindSystems(r.fs)
+	if err != nil {
+		return nil, err
+	}
+
+	resolvabilityDetails := make(map[string][]ResolvabilityDetails)
+	for _, system := range systems {
+		sid := filepath.Base(system)
+		profileFiles, err := sapsystem.FindProfiles(r.fs, sid)
+		if err != nil {
+			return nil, SapLocalhostResolverFileSystemError.Wrap(err.Error())
+		}
+
+		for _, profileFile := range profileFiles {
+			if profileFile == "DEFAULT.PFL" {
+				continue
+			}
+
+			match := hostnameRegexCompiled.FindStringSubmatch(profileFile)
+			if len(match) != len(subgroupMapping)+1 {
+				continue
+			}
+			addresses, err := r.hr.LookupHost(match[3])
+			if err != nil {
+				return nil, SapLocalhostResolverHostnameResolutionError.Wrap(err.Error())
+			}
+
+			details := ResolvabilityDetails{
+				Hostname:     match[3],
+				Addresses:    addresses,
+				InstanceName: match[2],
+			}
+			if _, ok := resolvabilityDetails[match[1]]; !ok {
```

Comment:
Could you use found, instead of ok? Ok is used more in typecasting scenarios, found is more appropriate if you are checking if you found something in the map